daemon/src/projection.rs

```diff
@@ -480,11 +492,13 @@ impl Cfd {
                 self.aggregated.cet = Some(cet);
 
                 self.aggregated.state = CfdState::PendingCet;
+                self.aggregated.has_unspent_joint_output = false;
             }
             RefundTimelockExpired { .. } => {
                 self.aggregated.refund_published = true;
 
                 self.aggregated.state = CfdState::PendingRefund;
+                self.aggregated.has_unspent_joint_output = false;
             }
             OracleAttestedPriorCetTimelock {
                 timelocked_cet,
```

Comment:
📯
Is it correct that we can achieve this if we publish the commit transaction too late and the CET timelock has not expired yet when the oracle attests to the price? 

I think this needs to be false as well. If the oracle attests to the price we know already the closing price.

Reply:
Hmm, interesting!

I guess we can say that having a valid but _timelocked_ transaction that can spend the joint output should also already account for the output being spent (which is what the boolean represents).

Reply:
> I think this needs to be false as well. If the oracle attests to the price we know already the closing price.

By itself, that reasoning does not make sense though. This boolean tracks whether we have an unspent join output, not whether we know the closing price. I think this comment highlights that we are introducing an unfit abstraction here.